pkg/kv/kvserver/replica_proposal.go

```diff
@@ -815,6 +815,25 @@ type proposalResult struct {
 	EndTxns            []result.EndTxnIntents
 }
 
+func makeProposalResult(
+	br *kvpb.BatchResponse, pErr *kvpb.Error, ei []roachpb.Intent, eti []result.EndTxnIntents,
+) proposalResult {
+	return proposalResult{
+		Reply:              br,
+		Err:                pErr,
+		EncounteredIntents: ei,
+		EndTxns:            eti,
+	}
+}
```

Comment:
Mild preference for just keeping this inlined. It's more readable considering the number of parameters.

Reply:
Noted - I was close to on the fence as well, but came down on the other side because the method forces all fields to be specified. There is a lot of ad-hoc / piecemeal initialisation going on in the proposal path, which I would like to see reduced.